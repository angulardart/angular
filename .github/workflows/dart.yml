# Created with package:mono_repo v5.0.2
name: Dart CI
on:
  push:
    branches:
      - master
      - google
  pull_request:
    branches:
      - master
      - google
defaults:
  run:
    shell: bash
env:
  PUB_ENVIRONMENT: bot.github

jobs:
  job_001:
    name: mono_repo self validate
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:stable"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: stable
      - id: checkout
        uses: actions/checkout@v2.3.4
      - name: mono_repo self validate
        run: dart pub global activate mono_repo 5.0.2
      - name: mono_repo self validate
        run: dart pub global run mono_repo generate --validate
  job_002:
    name: "analyze; PKG: angular_ast; `dart analyze`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:analyze_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: angular_ast; dart analyze
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: dart analyze
  job_003:
    name: "analyze; PKGS: angular_compiler, examples/hello_world; `dart analyze --fatal-infos`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler-examples/hello_world;commands:analyze_1"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler-examples/hello_world
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; dart analyze --fatal-infos"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart analyze --fatal-infos
      - id: examples_hello_world_pub_upgrade
        name: examples/hello_world; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart pub upgrade
      - name: "examples/hello_world; dart analyze --fatal-infos"
        if: "always() && steps.examples_hello_world_pub_upgrade.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart analyze --fatal-infos
  job_004:
    name: "build; PKG: _tests; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run build_runner build --fail-on-severe"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_005:
    name: "build; PKG: angular_ast; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: "angular_ast; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_006:
    name: "build; PKG: angular_compiler; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_007:
    name: "build; PKG: angular_forms; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_forms_pub_upgrade
        name: angular_forms; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_forms
        run: dart pub upgrade
      - name: "angular_forms; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_forms_pub_upgrade.conclusion == 'success'"
        working-directory: angular_forms
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_008:
    name: "build; PKG: angular_router; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_router_pub_upgrade
        name: angular_router; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_router
        run: dart pub upgrade
      - name: "angular_router; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_router_pub_upgrade.conclusion == 'success'"
        working-directory: angular_router
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_009:
    name: "build; PKG: angular_test; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_test_pub_upgrade
        name: angular_test; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_test
        run: dart pub upgrade
      - name: "angular_test; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_test_pub_upgrade.conclusion == 'success'"
        working-directory: angular_test
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_010:
    name: "build; PKG: examples/hello_world; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:examples/hello_world;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:examples/hello_world
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: examples_hello_world_pub_upgrade
        name: examples/hello_world; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart pub upgrade
      - name: "examples/hello_world; pub run build_runner build --fail-on-severe"
        if: "always() && steps.examples_hello_world_pub_upgrade.conclusion == 'success'"
        working-directory: examples/hello_world
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_011:
    name: "unit_test; PKG: _tests; `pub run build_runner test --fail-on-severe -- -P browser`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_2"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run build_runner test --fail-on-severe -- -P browser"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run build_runner test --fail-on-severe -- -P browser
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_012:
    name: "unit_test; PKG: _tests; `pub run test -P vm`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_1"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run test -P vm"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run test -P vm
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_013:
    name: "unit_test; PKG: angular_ast; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: "angular_ast; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_014:
    name: "unit_test; PKG: angular_compiler; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_015:
    name: "unit_test; PKG: angular_forms; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_forms_pub_upgrade
        name: angular_forms; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_forms
        run: dart pub upgrade
      - name: "angular_forms; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_forms_pub_upgrade.conclusion == 'success'"
        working-directory: angular_forms
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_016:
    name: "unit_test; PKG: angular_router; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_router_pub_upgrade
        name: angular_router; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_router
        run: dart pub upgrade
      - name: "angular_router; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_router_pub_upgrade.conclusion == 'success'"
        working-directory: angular_router
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_017:
    name: "unit_test; PKG: angular_test; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_test_pub_upgrade
        name: angular_test; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_test
        run: dart pub upgrade
      - name: "angular_test; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_test_pub_upgrade.conclusion == 'success'"
        working-directory: angular_test
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_018:
    name: Notify failure
    if: "(github.event_name == 'push' || github.event_name == 'schedule') && failure()"
    runs-on: ubuntu-latest
    steps:
      - run: |
          curl -H "Content-Type: application/json" -X POST -d \
            "{'text':'Build failed! ${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}'}" \
            "${CHAT_WEBHOOK_URL}"
        env:
          CHAT_WEBHOOK_URL: "${{ secrets.TEAM_CHAT_WEBHOOK_URL }}"
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
      - job_011
      - job_012
      - job_013
      - job_014
      - job_015
      - job_016
      - job_017
