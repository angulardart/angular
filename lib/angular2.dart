library angular2;

/**
 * An all-in-one place to import Angular 2 stuff.
 *
 * This library does not include `bootstrap`. Import `bootstrap.dart` instead.
 */
export 'package:angular2/core.dart';
export 'package:angular2/common.dart';
export 'package:angular2/instrumentation.dart';
export 'package:angular2/src/core/angular_entrypoint.dart'
    show AngularEntrypoint;
export 'package:angular2/src/core/application_tokens.dart'
    hide APP_ID_RANDOM_PROVIDER;
export 'package:angular2/src/platform/dom/dom_tokens.dart';
export 'package:angular2/src/platform/dom/dom_adapter.dart';
export 'package:angular2/src/platform/dom/events/event_manager.dart';
// Attention: Don't use angular2/src/compiler/compiler.dart
// for re exports as this will pull in the whole compiler,
// even if only some parts are shown.
// Background: Our transformer generates `initReflector` calls
// for all referenced modules, which will keep all
// types reachable from that module!
export 'package:angular2/src/compiler/url_resolver.dart';
export 'package:angular2/src/compiler/directive_resolver.dart';
export 'package:angular2/src/compiler/view_resolver.dart';
