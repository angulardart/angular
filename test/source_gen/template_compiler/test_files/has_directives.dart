import 'package:angular2/angular2.dart';

@Component(
    selector: 'test-foo',
    template: '<div>Foo</div>',
    directives: const [TestDirective, TestSubComponent]
    )
class TestFooComponent {}

@Directive(selector: 'directive')
class TestDirective {}

@Component(
    selector: 'test-bar',
    template: '<div>Bar</div>')
class TestSubComponent {}
