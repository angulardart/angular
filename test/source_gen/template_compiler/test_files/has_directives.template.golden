// GENERATED CODE - DO NOT MODIFY BY HAND

// **************************************************************************
// Generator: TemplateGenerator
// Target: library
// **************************************************************************

import 'has_directives.dart';
import 'package:angular2/src/core/reflection/reflection.dart' as _ngRef;
export 'has_directives.dart';
import 'package:angular2/src/core/render/api.dart';
import 'package:angular2/src/core/linker/app_view.dart';
import 'has_directives.dart' as import2;
import 'dart:html';
import 'package:angular2/src/core/di/injector.dart' as import4;
import 'package:angular2/src/core/linker/view_type.dart' as import5;
import 'package:angular2/src/core/change_detection/change_detection.dart';
import 'package:angular2/angular2.dart';
import 'package:angular2/src/core/linker/app_view_utils.dart' as import8;

const List<dynamic> styles_TestFooComponent = const [];
RenderComponentType renderType_TestFooComponent;

class ViewTestFooComponent0 extends AppView<import2.TestFooComponent> {
  Element _el_0;
  ViewTestFooComponent0(import4.Injector parentInjector,
      AppView<dynamic> parentView, num parentIndex, dynamic parentElement)
      : super(
            ViewTestFooComponent0,
            renderType_TestFooComponent,
            import5.ViewType.COMPONENT,
            {},
            parentInjector,
            parentView,
            parentIndex,
            parentElement,
            ChangeDetectionStrategy.CheckAlways);
  ComponentRef createInternal(dynamic rootSelector) {
    final Node parentRenderNode = initViewRoot(parentElement);
    var doc = document;
    _el_0 = doc.createElement('div');
    parentRenderNode.append(_el_0);
    Text _text_1 = new Text('Foo');
    _el_0.append(_text_1);
    init([], [_el_0, _text_1], []);
    return null;
  }
}

AppView<import2.TestFooComponent> viewFactory_TestFooComponent0(
    import4.Injector parentInjector,
    AppView<dynamic> parentView,
    num parentIndex,
    dynamic parentElement) {
  if (identical(renderType_TestFooComponent, null)) {
    (renderType_TestFooComponent = import8.appViewUtils
        .createRenderComponentType(
            '', 0, ViewEncapsulation.None, styles_TestFooComponent));
  }
  return new ViewTestFooComponent0(
      parentInjector, parentView, parentIndex, parentElement);
}

const List<dynamic> styles_TestFooComponentHost = const [];
RenderComponentType renderType_TestFooComponentHost;

class ViewTestFooComponentHost0 extends AppView<dynamic> {
  Element _el_0;
  AppView<import2.TestFooComponent> _compView_0;
  import2.TestFooComponent _TestFooComponent_0_2;
  ViewTestFooComponentHost0(import4.Injector parentInjector,
      AppView<dynamic> parentView, num parentIndex, dynamic parentElement)
      : super(
            ViewTestFooComponentHost0,
            renderType_TestFooComponentHost,
            import5.ViewType.HOST,
            {},
            parentInjector,
            parentView,
            parentIndex,
            parentElement,
            ChangeDetectionStrategy.CheckAlways);
  ComponentRef createInternal(dynamic rootSelector) {
    _el_0 = selectOrCreateHostElement('test-foo', rootSelector, null);
    _compView_0 =
        viewFactory_TestFooComponent0(this.injector(0), this, 0, _el_0);
    _TestFooComponent_0_2 = new import2.TestFooComponent();
    _compView_0.create(_TestFooComponent_0_2, projectableNodes, null);
    init([_el_0], [_el_0], []);
    return new ComponentRef(0, this, _el_0, _TestFooComponent_0_2);
  }

  dynamic injectorGetInternal(
      dynamic token, int requestNodeIndex, dynamic notFoundResult) {
    if ((identical(token, import2.TestFooComponent) &&
        (0 == requestNodeIndex))) {
      return _TestFooComponent_0_2;
    }
    return notFoundResult;
  }

  void detectChangesInternal() {
    _compView_0.detectChanges();
  }

  void destroyInternal() {
    _compView_0.destroy();
  }
}

AppView viewFactory_TestFooComponentHost0(import4.Injector parentInjector,
    AppView<dynamic> parentView, num parentIndex, dynamic parentElement) {
  if (identical(renderType_TestFooComponentHost, null)) {
    (renderType_TestFooComponentHost = import8.appViewUtils
        .createRenderComponentType(
            '', 0, ViewEncapsulation.Emulated, styles_TestFooComponentHost));
  }
  return new ViewTestFooComponentHost0(
      parentInjector, parentView, parentIndex, parentElement);
}

const ComponentFactory TestFooComponentNgFactory = const ComponentFactory(
    'test-foo',
    viewFactory_TestFooComponentHost0,
    import2.TestFooComponent,
    _METADATA);
const List<dynamic> styles_TestSubComponent = const [];
RenderComponentType renderType_TestSubComponent;

class ViewTestSubComponent0 extends AppView<import2.TestSubComponent> {
  Element _el_0;
  ViewTestSubComponent0(import4.Injector parentInjector,
      AppView<dynamic> parentView, num parentIndex, dynamic parentElement)
      : super(
            ViewTestSubComponent0,
            renderType_TestSubComponent,
            import5.ViewType.COMPONENT,
            {},
            parentInjector,
            parentView,
            parentIndex,
            parentElement,
            ChangeDetectionStrategy.CheckAlways);
  ComponentRef createInternal(dynamic rootSelector) {
    final Node parentRenderNode = initViewRoot(parentElement);
    var doc = document;
    _el_0 = doc.createElement('div');
    parentRenderNode.append(_el_0);
    Text _text_1 = new Text('Bar');
    _el_0.append(_text_1);
    init([], [_el_0, _text_1], []);
    return null;
  }
}

AppView<import2.TestSubComponent> viewFactory_TestSubComponent0(
    import4.Injector parentInjector,
    AppView<dynamic> parentView,
    num parentIndex,
    dynamic parentElement) {
  if (identical(renderType_TestSubComponent, null)) {
    (renderType_TestSubComponent = import8.appViewUtils
        .createRenderComponentType(
            '', 0, ViewEncapsulation.None, styles_TestSubComponent));
  }
  return new ViewTestSubComponent0(
      parentInjector, parentView, parentIndex, parentElement);
}

const List<dynamic> styles_TestSubComponentHost = const [];
RenderComponentType renderType_TestSubComponentHost;

class ViewTestSubComponentHost0 extends AppView<dynamic> {
  Element _el_0;
  AppView<import2.TestSubComponent> _compView_0;
  import2.TestSubComponent _TestSubComponent_0_2;
  ViewTestSubComponentHost0(import4.Injector parentInjector,
      AppView<dynamic> parentView, num parentIndex, dynamic parentElement)
      : super(
            ViewTestSubComponentHost0,
            renderType_TestSubComponentHost,
            import5.ViewType.HOST,
            {},
            parentInjector,
            parentView,
            parentIndex,
            parentElement,
            ChangeDetectionStrategy.CheckAlways);
  ComponentRef createInternal(dynamic rootSelector) {
    _el_0 = selectOrCreateHostElement('test-bar', rootSelector, null);
    _compView_0 =
        viewFactory_TestSubComponent0(this.injector(0), this, 0, _el_0);
    _TestSubComponent_0_2 = new import2.TestSubComponent();
    _compView_0.create(_TestSubComponent_0_2, projectableNodes, null);
    init([_el_0], [_el_0], []);
    return new ComponentRef(0, this, _el_0, _TestSubComponent_0_2);
  }

  dynamic injectorGetInternal(
      dynamic token, int requestNodeIndex, dynamic notFoundResult) {
    if ((identical(token, import2.TestSubComponent) &&
        (0 == requestNodeIndex))) {
      return _TestSubComponent_0_2;
    }
    return notFoundResult;
  }

  void detectChangesInternal() {
    _compView_0.detectChanges();
  }

  void destroyInternal() {
    _compView_0.destroy();
  }
}

AppView viewFactory_TestSubComponentHost0(import4.Injector parentInjector,
    AppView<dynamic> parentView, num parentIndex, dynamic parentElement) {
  if (identical(renderType_TestSubComponentHost, null)) {
    (renderType_TestSubComponentHost = import8.appViewUtils
        .createRenderComponentType(
            '', 0, ViewEncapsulation.Emulated, styles_TestSubComponentHost));
  }
  return new ViewTestSubComponentHost0(
      parentInjector, parentView, parentIndex, parentElement);
}

const ComponentFactory TestSubComponentNgFactory = const ComponentFactory(
    'test-bar',
    viewFactory_TestSubComponentHost0,
    import2.TestSubComponent,
    _METADATA);
const _METADATA = const <dynamic>[
  TestFooComponent,
  const <dynamic>[],
  TestDirective,
  const <dynamic>[],
  TestSubComponent,
  const <dynamic>[]
];
var _visited = false;
void initReflector() {
  if (_visited) {
    return;
  }
  _visited = true;
  _ngRef.reflector.registerType(
      TestFooComponent,
      new _ngRef.ReflectionInfo(
          const <dynamic>[], const [], () => new TestFooComponent()));
  _ngRef.reflector.registerType(
      TestDirective,
      new _ngRef.ReflectionInfo(
          const <dynamic>[], const [], () => new TestDirective()));
  _ngRef.reflector.registerType(
      TestSubComponent,
      new _ngRef.ReflectionInfo(
          const <dynamic>[], const [], () => new TestSubComponent()));
}
